### DISCUSSION TITLE
How to map multiple properties to one?

### DISCUSSION THREAD

**Discussion by Hantick** [riok/mapperly#1269] [Q&A] [answered]
How can i set up the mapping of 3 source properties to one destination property that has `required`
```csharp
public class DestinationItem 
{
    public required bool IsAccessible { get; set; }
}
```
```csharp
public class BasicItem 
{
    public bool? IsLocked { get; set; }
    public bool? ErrorMessage { get; set; }
    public bool? IsBlocked{ get; set; }
}
```
I want something like this
`[MapProperty([nameof(BasicItem.IsLocked), nameof(BasicItem.ErrorMessage ), nameof(BasicItem.IsBlocked)], [nameof(BookingListItem.IsAccessible)], Use = nameof(MapIsAccessible))]`
or
`[MapProperty(nameof(BasicItem), [nameof(BookingListItem.IsAccessible)], Use = nameof(MapIsAccessible))]`
```csharp
[UserMapping]
private static bool MapIsAccessible(BasicItem item)
    => (!item.ErrorMessage ?? true) && (!item.IsLocked ?? true) && (!item.IsBlocked?? true);
```

So basically all those 3 properties are logically combined to `IsAccessible` property

**Comment by latonz** [comment, marked as answer]
Starting with Mapperly `3.6.0-next.1` you can use `MapPropertyFromSource` exactly for this: [docs](https://next.mapperly.riok.app/docs/configuration/mapper/#mapping-from-source).
Would look sth. like this for you: `[MapPropertyFromSource(nameof(DestinationItem.IsAccessible), Use = nameof(MapIsAccessible))]`.

However it may be easier to just introduce a get only property `IsAccessible` on your `BasicItem`.

**Reply by hansmbakker** [reply]
Is this also the recommended way to map fallback values?

Imagine having a objects like this, where the goal is to have `destinationItem.Operator = sourceItem.Operator ?? sourceItem.Owner`.

```csharp
public class DestinationItem
{
    public object Operator  { get; set; }
    public object Owner { get; set; }
}

public class SourceItem 
{
    public object? Operator { get; set; }
    public object Owner { get; set; }
}
```

**Reply by latonz** [reply]
Yes yoiu can implement fallbacks like this.
